src/basic/fs-util.c

```diff
@@ -926,6 +926,7 @@ int chase_symlinks(const char *path, const char *original_root, unsigned flags,
                 if (fstat(child, &st) < 0)
                         return -errno;
                 if ((flags & CHASE_SAFE) &&
+                    (empty_or_root(root) || (unsigned)(todo - buffer) > strlen(root)) &&
```

Comment:
strlen() returns size_t btw, hence the left-hand cast should be to `(size_t)` too